sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/SaveAsHiveFile.scala

```diff
@@ -68,6 +68,26 @@ private[hive] trait SaveAsHiveFile extends DataWritingCommand {
         .get("mapreduce.output.fileoutputformat.compress.type"))
     }
 
+    fileSinkConf.tableInfo.getOutputFileFormatClassName match {
+      case formatName if formatName.endsWith("ParquetOutputFormat") =>
+        val compressionConf = "parquet.compression"
+        val compressionCodec = getCompressionByPriority(fileSinkConf, compressionConf,
+          sparkSession.sessionState.conf.parquetCompressionCodec) match {
+          case "NONE" => "UNCOMPRESSED"
+          case _@x => x
+        }
+        hadoopConf.set(compressionConf, compressionCodec)
+      case formatName if formatName.endsWith("OrcOutputFormat") =>
+        val compressionConf = "orc.compress"
+        val compressionCodec = getCompressionByPriority(fileSinkConf, compressionConf,
+          sparkSession.sessionState.conf.orcCompressionCodec) match {
+          case "UNCOMPRESSED" => "NONE"
+          case _@x => x
```

Comment:
`case x => x`?

Reply:
In fact, the following process will check the correctness of this value, and because "orcoptions" is not accessable here, I have to add the "uncompressed" => "NONE" conversion.
Do you have any good advice?